go/vt/vtgate/scatter_conn.go

```diff
@@ -212,6 +217,18 @@ func (stc *ScatterConn) ExecuteMultiShard(
 	return qr, err
 }
 
+func (stc *ScatterConn) executeAutocommit(ctx context.Context, target *querypb.Target, sql string, bindVariables map[string]*querypb.BindVariable, options *querypb.ExecuteOptions) (*sqltypes.Result, error) {
+	queries := []*querypb.BoundQuery{{
+		Sql:           sql,
+		BindVariables: bindVariables,
+	}}
+	qrs, err := stc.gateway.ExecuteBatch(ctx, target, queries, true /* asTransaction */, 0, options)
```

Comment:
I think it would be worth capturing in a comment that our use of ExecuteBatch is a stopgap because there is not yet a good way to do autocommit with a regular Execute.